PocketKit/Sources/Sync/PocketSaveService.swift

```diff
@@ -219,9 +219,11 @@ class SaveOperation<Mutation: GraphQLMutation>: AsyncOperation {
     }
 
     private func storeUnresolvedSavedItem() {
-        let unresolved: UnresolvedSavedItem = space.new()
-        unresolved.savedItem = savedItem
-        try? space.save()
```

Comment:
Should this method always save on the main thread under the hood?

Reply:
Yes, but so do all of the other methods on `space` (e.g. `new()`). hence wrapping them in `performAndWait` here instead of encapsulating it in the actual `save` method.